tests/lib/rules/manifest-app-name/tests.ts

```diff
@@ -0,0 +1,90 @@
+import { generateHTMLPage } from '../../../helpers/misc';
+import { getRuleName } from '../../../../src/lib/utils/rule-helpers';
+import { IRuleTest } from '../../../helpers/rule-test-type'; // eslint-disable-line no-unused-vars
+import * as ruleRunner from '../../../helpers/rule-runner';
+
+const htmlWithManifestSpecified = generateHTMLPage('<link rel="manifest" href="site.webmanifest">');
+
+const tests: Array<IRuleTest> = [
+    {
+        name: `Manifest is not specified`,
+        serverConfig: generateHTMLPage('<link rel="stylesheet" href="style.css">')
+    },
+    {
+        name: `Manifest is specified and its content is not valid JSON`,
+        serverConfig: {
+            '/': htmlWithManifestSpecified,
+            '/site.webmanifest': 'x'
+        }
+    },
+    {
+        name: `Manifest is specified without 'name' and 'short_name'`,
+        reports: [{ message: `Manifest should contain the 'name' member` }],
+        serverConfig: {
+            '/': htmlWithManifestSpecified,
+            '/site.webmanifest': { content: '{}' }
+        }
+    },
+    {
+        name: `Manifest is specified with empty 'name' and no 'short_name'`,
+        reports: [{ message: `Manifest should contain non-empty 'name' member` }],
+        serverConfig: {
+            '/': htmlWithManifestSpecified,
+            '/site.webmanifest': { content: `{ "name": "     " }` }
+        }
+    },
+    {
+        name: `Manifest is specified with 'name' and no 'short_name'`,
+        serverConfig: {
+            '/': htmlWithManifestSpecified,
+            '/site.webmanifest': { content: `{ "name": "🐛🐛🐛🐛🐛🐛🐛🐛🐛🐛🐛🐛" }` }
+        }
+    },
+    {
+        name: `Manifest is specified with long 'name' and 'short_name'`,
+        reports: [{ message: `Manifest should have 'name' member under 30 characters` }],
+        serverConfig: {
+            '/': htmlWithManifestSpecified,
+            '/site.webmanifest': { content: `{ "name": "1234567890123456789012345678901", "short_name": "test" }` }
+        }
+    },
+    {
+        name: `Manifest is specified with 'name' and 'short_name'`,
+        serverConfig: {
```

Comment:
Shouldn't this test fail? `name.length` is <= 12

I don't see any `reports`

Reply:
No. If `name` is up to 12, then it can be used instead of`short_name. However, is `short_name` is also specified, it's perfectly valid, and not a problem.

Reply:
This point of the documentation makes me think it should fail:

> If the `name` is under or 12 characters, there isn't a need to specify `short_name` as browsers can and will just use `name`.

If they will just use `name` if less than 12 characters, then `short_name` shouldn't be there. If they can decide what to use depending on the context and the space (list of apps, title below and icon) then I think this phrase should be reformulated.

Reply:
> then I think this phrase should be reformulated.

Any suggestions?

Reply:
Done.